### ISSUE TITLE
[UI] Can't set cursor image

### ISSUE THREAD

**Issue by darrylryan** [AtomicGameEngine/AtomicGameEngine#1627]
The Urho UI provides a method to set the cursor image however Turbobadger doesn't offer this so this functionality went away with SystemUI.

**Comment by Alan-FGR** [comment]
I'm also interested in this... I don't know how the UIs do that, but SDL offers world class xplat cursor image support, including hardware cursors, don't know whether it's available in scripting though.

**Comment by Alan-FGR** [comment]
Wen you set the hardware cursor using SDL, apparently there's no restriction on image size, what strikes me as a but odd to say the least and indeed in my hardware any size seems to work perfectly fine, some engine however restrict hardware cursor to very small sizes (e.g.:32x32px) and fallback to the terrible software cursor when you exceed that size, I couldn't find any reason regarding that limitation on the hardware side though, SDL seems to quite happily use any size, but I'm afraid in some hardware/platforms that'll break badly. Any ideas on this?

**Comment by mattbenic** [comment]
So what is the current issue? Is the ability to set the size just not exposed (since you say you were able to set it)

**Comment by Alan-FGR** [comment]
Nope, you can set it using SDL directly, there's no API and I've no idea on how the UIs set that, Urho UI probably just abstracts the SDL stuff, no idea on TB, but if it's software cursor that's not good, hardware cursor is the way to go but there gotta be some safe limits otherwise other engines won't be fallbacking.

**Comment by rezasailo** [comment]
Tank
